model/entry.go

```diff
@@ -0,0 +1,75 @@
+package model
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/prometheus/common/model"
+)
+
+type Entry struct {
+	Ts   time.Time
+	Line string
+}
+
+// MarshalJSON implements the json.Marshaler interface.
+func (e *Entry) MarshalJSON() ([]byte, error) {
+	l, err := json.Marshal(e.Line)
+	if err != nil {
+		return nil, err
+	}
+	return []byte(fmt.Sprintf("[\"%d\",%s]", e.Ts.UnixNano(), l)), nil
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (e *Entry) UnmarshalJSON(data []byte) error {
+	var unmarshal []string
+
+	err := json.Unmarshal(data, &unmarshal)
+	if err != nil {
+		return err
+	}
+
+	t, err := strconv.ParseInt(unmarshal[0], 10, 64)
+	if err != nil {
+		return err
+	}
+
+	e.Ts = time.Unix(0, t)
+	e.Line = unmarshal[1]
+
+	return nil
+}
+
+type IntEntry struct {
```

Comment:
I think it would be good to use either `Ts` as an `int` or as `time.Time`. I think the argument for `int` may primarily be efficiency/space reasons, but it's probably negligible and we can just use `time.Time`. Point being, I believe all entries should use the _same_ data structure here so we don't have some that use `int` and some that use `time.Time.` 

@cyriltovena @slim-bean @rfratto Do y'all agree?

Reply:
@CyrilPeponnet @slim-bean @rfratto any update ?

Reply:
I think it would be better if we can achieve what @owen-d suggest, but I'd just be careful about hot path, worth running benchmark before and after.